PurchasesCoreSwift/Caching/InMemoryCachedObject.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  InMemoryCachedObject.swift
+//  PurchasesCoreSwift
+//
+//  Created by Joshua Liebowitz on 7/13/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+
+open class InMemoryCachedObject<T> {
+
+    private let accessQueue = DispatchQueue(label: "InMemoryCachedObjectQueue", attributes: .concurrent)
+    private var lastUpdated: Date?
+    private var cachedObject: T?
+    public var lastUpdatedAt: Date? {
+        accessQueue.sync {
+            return lastUpdated
+        }
+    }
+
+    public init() { }
+
+    open func isCacheStale(durationInSeconds: Double) -> Bool {
+        accessQueue.sync {
+            guard let lastUpdated = lastUpdated else {
+                return true
+            }
+
+            let timeSinceLastCheck = -1.0 * lastUpdated.timeIntervalSinceNow
+            return timeSinceLastCheck >= durationInSeconds
+        }
+    }
+
+    open func clearCacheTimestamp() {
+        accessQueue.sync(flags: .barrier) {
+            self.lastUpdated = nil
+        }
+    }
+
+    open func clearCache() {
+        accessQueue.sync(flags: .barrier) {
+            self.lastUpdated = nil
+            self.cachedObject = nil
+        }
+    }
+
+    open func updateCacheTimestamp(date: Date) {
```

Comment:
why do these need to be `open`? if it's just for testing purposes, making them `internal` and using `@testable` imports should work

Reply:
Same as other open questions, I forgot 🤦